python/refactor/nn/modules/encoder.py

```diff
@@ -0,0 +1,138 @@
+from .module import Module
+from .ke_embedding import KGEmbedding
+import torch as th
+import itertools
+
+# this is an interface so it will not have many parameters to be defined.
+class Encoder(Module):
+    def __init__(self, encoder_name):
+        super(Encoder, self).__init__(encoder_name)
+
+    def forward(self, data, gpu_id):
+        raise NotImplementedError
+
+class KGEEncoder(Encoder):
+    def __init__(self,
+                 hidden_dim,
+                 lr,
+                 optimizer,
+                 n_entity,
+                 n_relation,
+                 init_func,
+                 rel_parts=None, # ! rel_parts and cross_rels must be put here otherwise harm the general structure
+                 cross_rels=None,
+                 soft_rel_part=False,
+                 strict_rel_part=False,
```

Comment:
we'll move most of the arguments as the arguments of the fit function.

Reply:
I delete most of the arguments as it's for multi-GPU.